### PR TITLE
feat(core): log error body

### PR DESCRIPTION
**Pull request by IceHe**
## Summary

Save the error context from `RequestError.body` in logs

See Slack thread [discussion](https://silverhand-io.slack.com/archives/C02A8G4HVAM/p1654570553888139)

- [LOG-2875 : Save RequestError.data in the logs](https://linear.app/silverhand/issue/LOG-2875/save-requesterrordata-in-the-logs)

## Linear Issue Reference

LOG-2875

## Testing

image

### CODE REVIEW

**Review by wangsijie — APPROVED** [review]

File: packages/core/src/middleware/koa-log.test.ts [hunk 1]
```diff
@@ -60,33 +62,69 @@ describe('koaLog middleware', () => {
...
+      const ctx: WithLogContext<ReturnType<typeof createContextWithRouteParameters>> = {
+        ...createContextWithRouteParameters({ headers: { 'user-agent': userAgent } }),
+        // Bypass middleware context type assert
+        addLogContext,
+        log,
+      };
+      ctx.request.ip = ip;
+
+      const message = 'normal error';
+      const error = new Error(message);
```
Comment (darcyYe) [inline packages/core/src/middleware/koa-log.test.ts:76]: Can write in a single line.

File: packages/core/src/middleware/koa-log.test.ts [hunk 2]
```diff
@@ -60,33 +62,69 @@ describe('koaLog middleware', () => {
...
+        ctx.log(type, payload);
+        throw error;
+      };
+      await expect(koaLog()(ctx, next)).rejects.toMatchError(error);
+
+      expect(insertLogMock).toBeCalledWith({
+        id: nanoIdMock,
+        type,
+        payload: {
+          ...payload,
```
Comment (darcyYe) [inline packages/core/src/middleware/koa-log.test.ts:88]: This `payload` is a little bit confusing to me, was thinking this `payload` is `logger`'s property.
Maybe using `mockPayload` or something for better understanding.

File: packages/core/src/middleware/koa-log.ts [hunk 3]
```diff
@@ -90,7 +91,11 @@ export default function koaLog<
     try {
       await next();
     } catch (error: unknown) {
-      logger.set({ result: LogResult.Error, error: String(error) });
+      const errorInfo = error instanceof RequestError ? JSON.stringify(error.body) : String(error);
+      logger.set({
+        result: LogResult.Error,
+        error: errorInfo,
```
Comment (gao-sun) [inline packages/core/src/middleware/koa-log.ts:97]: do we log error code and message?

File: packages/core/src/middleware/koa-log.test.ts [hunk 4]
```diff
@@ -60,33 +62,69 @@ describe('koaLog middleware', () => {
...
+      const ctx: WithLogContext<ReturnType<typeof createContextWithRouteParameters>> = {
+        ...createContextWithRouteParameters({ headers: { 'user-agent': userAgent } }),
+        // Bypass middleware context type assert
+        addLogContext,
+        log,
+      };
+      ctx.request.ip = ip;
+
+      const errorMessage = 'Error message';
+      jest.spyOn(i18next, 't').mockReturnValue(errorMessage);
```
Comment (darcyYe) [inline packages/core/src/middleware/koa-log.test.ts:107]: Prefer `mockReturnValueOnce` than `mockReturnValue`.

File: packages/core/src/middleware/koa-log.test.ts [hunk 5]
```diff
@@ -60,33 +62,69 @@ describe('koaLog middleware', () => {
...
+        ctx.log(type, payload);
+        throw error;
+      };
+      await expect(koaLog()(ctx, next)).rejects.toMatchError(error);
+
+      expect(insertLogMock).toBeCalledWith({
+        id: nanoIdMock,
+        type,
+        payload: {
+          ...payload,
```
Comment (darcyYe) [inline packages/core/src/middleware/koa-log.test.ts:121]: Same here.

File: packages/core/src/middleware/koa-log.ts (hunk 3 above)
Comment (IceHe) [reply packages/core/src/middleware/koa-log.ts:97]: Have added `error code` and `message` in the error logs in this PR.

Maybe need further discussion?

Comment (IceHe) [reply packages/core/src/middleware/koa-log.ts:97]: Refactored.

See Slack thread [discussion](https://silverhand-io.slack.com/archives/C02A8G4HVAM/p1654590201727249?thread_ts=1654570553.888139&cid=C02A8G4HVAM)

**Pull request merged**